### PR TITLE
snap: enforce minimal snap name len of 2

### PR DESCRIPTION
**Pull request by mvo5**
Trivial drive-by after looking at LP: #1763048

### CODE REVIEW

File: snap/validate.go [hunk 1]
```diff
@@ -93,7 +93,7 @@ func ValidateInstanceName(instanceName string) error {
 func ValidateName(name string) error {
 	// NOTE: This function should be synchronized with the two other
 	// implementations: sc_snap_name_validate and validate_snap_name .
```
Comment (stolowski) [inline snap/validate.go]: Per this comment, it seems we need similiar change in sc_snap_name_validate?

File: cmd/libsnap-confine-private/snap-test.c [hunk 2]
```diff
@@ -201,6 +201,8 @@ static void test_sc_snap_or_instance_name_validate(gconstpointer data)
 	const char *invalid_names[] = {
 		// name cannot be empty
 		"",
+		// too short
+		"a",
```
Comment (zyga) [inline cmd/libsnap-confine-private/snap-test.c]: Nice touch :)

File: cmd/libsnap-confine-private/snap-test.c [hunk 3]
```diff
@@ -317,8 +319,8 @@ static void test_sc_instance_name_validate(void)
 	sc_error_free(err);
 
 	const char *valid_names[] = {
-		"a", "aa", "aaa", "aaaa",
-		"a_a", "aa_1", "a_123", "a_0123456789",
```
Comment (zyga) [inline cmd/libsnap-confine-private/snap-test.c]: Interesting that short snap name with long instance key is still too short. Useful test.

File: cmd/snap-update-ns/bootstrap.c [hunk 4]
```diff
@@ -239,6 +239,10 @@ int validate_snap_name(const char *snap_name)
 		bootstrap_msg = "snap name must contain at least one letter";
 		return -1;
 	}
+        if (n < 2) {
```
Comment (zyga) [inline cmd/snap-update-ns/bootstrap.c]: Formatting seems off, I'll correct that.

File: snap/validate.go [hunk 5]
```diff
@@ -93,7 +93,7 @@ func ValidateInstanceName(instanceName string) error {
...
-	if len(name) > 40 || !isValidName(name) {
+	if len(name) < 2 || len(name) > 40 || !isValidName(name) {
 		return fmt.Errorf("invalid snap name: %q", name)
```
Comment (zyga) [inline snap/validate.go]: I find it funny that obscure C code has better error messages than the main go code. Not something for this PR to fix though.

**Pull request merged**